pkg/loadbalancer/flow/nfthandler.go

```diff
@@ -0,0 +1,475 @@
+			&expr.Cmp{
+				Op:       expr.CmpOpEq,
+				Register: 1,
+				Data:     encodePort(dport),
+			},
+			// [ immediate reg 1 0x00008913 ] 1389 = 5001
+			&expr.Immediate{
+				Register: 1,
+				Data:     encodePort(localPort),
+			},
+			// [ payload write reg 1 => 2b @ transport header + 2 csum_type 1 csum_off 16 csum_flags 0x0 ]
+			&expr.Payload{
+				OperationType:  expr.PayloadWrite,
+				SourceRegister: 1,
+				Base:           expr.PayloadBaseTransportHeader,
+				Offset:         2,
+				Len:            2,
+				CsumType:       expr.CsumTypeInet,
+				CsumOffset:     16,
+			},
+			// [ counter pkts 0 bytes 0 ]
+			&expr.Counter{
+				Bytes:   0,
+				Packets: 0,
+			},
+			// [ notrack ]
+			&expr.Notrack{},
+		},
+	}
+}
+
+// buildSrcRule builds a Rule for sport NAT (egress)
+func (nfth *nfth) buildSrcRule(
+	ch *nftables.Chain, l3Proto, l4Proto byte, set *nftables.Set, sport, localPort uint) *nftables.Rule {
+
+	var adrLen, adrOffset uint32
+	if l3Proto == unix.AF_INET {
+		adrLen = 4
+		adrOffset = 12
+	} else {
+		adrLen = 16
+		adrOffset = 8
+	}
+
+	return &nftables.Rule{
+		Table: nfth.table,
+		Chain: ch,
+		Exprs: []expr.Any{
+			// [ meta load nfproto => reg 1 ]
+			&expr.Meta{
+				Key:      expr.MetaKeyNFPROTO,
+				Register: 1,
+			},
+			// [ cmp eq reg 1 0x00000002 ]
+			&expr.Cmp{
+				Op:       expr.CmpOpEq,
+				Register: 1,
+				Data:     []byte{l3Proto},
+			},
+			// [ payload load 4b @ network header + 12 => reg 1 ]
+			&expr.Payload{
+				DestRegister: 1,
+				Base:         expr.PayloadBaseNetworkHeader,
+				Offset:       adrOffset,
+				Len:          adrLen,
+			},
+			// [ lookup reg 1 set vips ]
+			&expr.Lookup{
+				SourceRegister: 1,
+				SetName:        set.Name,
+				SetID:          set.ID,
+			},
+			// [ meta load l4proto => reg 1 ]
+			&expr.Meta{
+				Key:      expr.MetaKeyL4PROTO,
+				Register: 1,
+			},
+			// [ cmp eq reg 1 0x00000006 ]
+			&expr.Cmp{
+				Op:       expr.CmpOpEq,
+				Register: 1,
+				Data:     []byte{l4Proto},
+			},
+			// [ payload load 2b @ transport header + 0 => reg 1 ]
+			&expr.Payload{
+				DestRegister: 1,
+				Base:         expr.PayloadBaseTransportHeader,
+				Offset:       0,
+				Len:          2,
+			},
+			// [ cmp eq reg 1 0x0000611e ] 1e61 = 7777
+			&expr.Cmp{
+				Op:       expr.CmpOpEq,
+				Register: 1,
+				Data:     encodePort(localPort),
+			},
+			// [ immediate reg 1 0x00008913 ] 1389 = 5001
+			&expr.Immediate{
+				Register: 1,
+				Data:     encodePort(sport),
+			},
+			// [ payload write reg 1 => 2b @ transport header + 0 csum_type 1 csum_off 16 csum_flags 0x0 ]
+			&expr.Payload{
+				OperationType:  expr.PayloadWrite,
+				SourceRegister: 1,
+				Base:           expr.PayloadBaseTransportHeader,
+				Offset:         0,
+				Len:            2,
+				CsumType:       expr.CsumTypeInet,
+				CsumOffset:     16,
+			},
+			// [ counter pkts 0 bytes 0 ]
+			&expr.Counter{
+				Bytes:   0,
+				Packets: 0,
+			},
+			// [ notrack ]
+			&expr.Notrack{},
```

Comment:
The chain prio is 100, but AFAIK notrack must be added before ChainPriorityConntrack to take effect

Reply:
Does that apply when the purpose is not to prevent packet reassembly?

The reason here applies for *all* traffic through the Meridio LB; since return traffic may take another path the conntracker may never see the complete connection. So connection tracking is pointless.

The Meridio LB should perhaps have a general `notrack` in a high prio chain, but that is not a responsibility in this PR. I can remove Notrack, but I will not re-prioritize the chain.

Reply:
Left as-is for now.